metaflow/metadata/metadata.py

```diff
@@ -513,8 +513,28 @@ def _tags(self):
             tags.append("r_version:" + env["r_version_code"])
         return tags
 
-    def _register_code_package_metadata(self, run_id, step_name, task_id, attempt):
-        metadata = []
+    def _register_system_metadata(self, run_id, step_name, task_id, attempt):
+        env = self._environment.get_environment_info()
```

Comment:
Could we, instead of doing this, rely on the information coming from `self._tags`. The reason for this is that internally I extend `_tags` and that would allow me to reuse this code as is (ie: not do much else than call this function) to get all the proper tags in place as metadata. As a transition (if we keep both for a while), we can just strip out the date here. The identity can also be stripped out though I suppose it would be nice to be added the run.

I am also curious as to why we want to keep things that are non-varying in the metadata (like the r information). My thought was that we could:
- keep all current system tags that are non-varying as system tags at the run level
- keep things that can vary across tasks/steps as metadata (like the python version for example)